test/ctia/test_helpers/access_control.clj

```diff
@@ -951,86 +951,110 @@
 
 (defn test-access-control-tlp-settings
-  [entity new-entity]
-  (testing "TLP Settings Enforcement"
-    (swap! (p/global-properties-atom) assoc-in
-           [:ctia :access-control]
-           {:default-tlp "amber"
-            :min-tlp "amber"})
-
-    (let [{status-default-tlp :status
-           body-default-tlp :parsed-body}
-          (post (format "ctia/%s" entity)
-                :body (dissoc new-entity :tlp)
-                :headers {"Authorization" "player-1-token"})
-          {status-disallowed-tlp :status
-           body-disallowed-tlp :parsed-body}
-          (post (format "ctia/%s" entity)
-                :body (assoc new-entity :tlp "white")
-                :headers {"Authorization" "player-1-token"})]
-
-      (is (= 201 status-default-tlp))
-      (is (= "amber" (:tlp body-default-tlp)))
-
-
-      (is (= 400 status-disallowed-tlp))
-      (is (= "Invalid document TLP white, allowed TLPs are: amber,red"
-             (:message body-disallowed-tlp))))))
-
+  [entity new-entity fixtures]
+  (helpers/with-config-transformer*
+    #(assoc-in % [:ctia :access-control] {:default-tlp "amber"
+                                          :min-tlp "amber"})
+    #(fixtures
+       (fn []
+         (testing "TLP Settings Enforcement"
+           (let [;; verify the with-config-transformer* call above--it's possible
+                 ;; a fixture could override it since we call it first
+                 get-in-config (helpers/current-get-in-config-fn)
+                 _ (assert (= (get-in-config [:ctia :access-control])
+                              {:default-tlp "amber"
+                               :min-tlp "amber"})
+                           (get-in-config [:ctia :access-control]))
+                 
+                 {status-default-tlp :status
+                  body-default-tlp :parsed-body}
+                 (post (format "ctia/%s" entity)
+                       :body (dissoc new-entity :tlp)
+                       :headers {"Authorization" "player-1-token"})
+                 {status-disallowed-tlp :status
+                  body-disallowed-tlp :parsed-body}
+                 (post (format "ctia/%s" entity)
+                       :body (assoc new-entity :tlp "white")
+                       :headers {"Authorization" "player-1-token"})]
+
+             (is (= 201 status-default-tlp))
+             (is (= "amber" (:tlp body-default-tlp)))
+
+
+             (is (= 400 status-disallowed-tlp))
+             (is (= "Invalid document TLP white, allowed TLPs are: amber,red"
+                    (:message body-disallowed-tlp)))))))))
+
+;; the body of this function must change the current TK config
+;; via test-access-control-tlp-settings. This is only possible by starting a new app.
+;; Since apps are started by fixtures, this function accepts a fixture and 
+;; must not be called in a context wrapped by fixture-ctia (this is dynamically enforced).
 (defn access-control-test
```

Comment:
The changes below are mostly a reshuffling of existing code.

We redefined `fixtures` to include the setup of capabilities and such, then we separate `test-access-control-tlp-settings` from the rest of the tests because it needs to set the config before starting CTIA.